File: REVENGG/hehe_saksham/decoder.py
# Emoji Cipher Decoder (Text-Based Emojis)

# Mapping of unique text emojis to letters
EMOJI_TO_LETTER = {
    '(ᵔᴥᵔ)': 'A', '(•ω•)': 'B', '(◕‿◕)': 'C', '(≧▽≦)': 'D', '(＾▽＾)': 'E',
    '(¬‿¬)': 'F', '(◠‿◠)': 'G', '(✿◠‿◠)': 'H', '(◕‿◕✿)': 'I', '(◠ω◠)': 'J',
    '(◠△◠)': 'K', '(◠‿◠✿)': 'L', '(◠﹏◠)': 'M', '(◠‿◠)ノ': 'N', '(◠‿◠✿)ノ': 'O',
    '(◠‿◠)♡': 'P', '(◠‿◠✿)♡': 'Q', '(◠‿◠)♫': 'R', '(◠‿◠✿)♫': 'S', '(◠‿◠)☼': 'T',
    '(◠‿◠✿)☼': 'U', '(◠‿◠)☁': 'V', '(◠‿◠✿)☁': 'W', '(◠‿◠)✧': 'X', '(◠‿◠✿)✧': 'Y',
    '(◠‿◠)★': 'Z', '␣': ' '
}

def decode_emoji_cipher(encoded_text):
    """
    Decodes an emoji-encoded message back into plain text using unique text emojis.
    """
    decoded_text = []
    # Split the encoded text into individual emojis
    for emoji in encoded_text.split(' '):
        if emoji in EMOJI_TO_LETTER:
            decoded_text.append(EMOJI_TO_LETTER[emoji])
        else:
            decoded_text.append('')  # Skip unknown emojis
    return ''.join(decoded_text)

if __name__ == "__main__":
    # Input the encoded message
    encoded_message = input("Enter the encoded emoji cipher: ")
    
    # Decode the message
    decoded_message = decode_emoji_cipher(encoded_message)
    
    # Output the decoded text
    print("Decoded Message:")
    print(decoded_message)

File: REVENGG/Q1/decoder.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

int Decoding(string& encoded, vector<int>& fibb) {
    int index = fibb.size() - 2;
    int num = 0;
    if (encoded.size() < 2 || encoded.back() != '0' || encoded[encoded.size() - 2] != '0')    return -1;
    for (int i = 0; i < encoded.size() - 1; ++i) {
        if (encoded[i] == '0')    num += fibb[i];
        else if (encoded[i] != '1')    return -1;
    }
    return num;
}

int main() {
    string encoded;
    cout<<"Enter Encoded Input: "<<endl;
    cin>>encoded;
    vector<int> fibb = {1, 2};
    while (fibb.size() <= encoded.size() - 2)    fibb.push_back(fibb[fibb.size() - 1] + fibb[fibb.size() - 2]);
    int decoded_number = Decoding(encoded, fibb);
    if (decoded_number != -1)    cout<<"Decoded Output: "<<decoded_number<<endl;
    return 0;
}